server/src/main/java/org/opensearch/identity/ExtensionTokenProcessor.java

```diff
@@ -44,19 +121,37 @@ public PrincipalIdentifierToken generateToken(Principal principal) {
      *
      * @opensearch.internal
      *
-     * This method contains a placeholder implementation.
+     * This method contains a placeholder simplementation.
      * More concrete implementation will be covered in https://github.com/opensearch-project/OpenSearch/issues/4485
      */
     public Principal extractPrincipal(PrincipalIdentifierToken token) throws IllegalArgumentException {
```

Comment:
Can you write tests for this to shows its usage?

Reply:
yeah, I was just hoping to put this out there as soon as possible because I know we are trying to push rapidly & everyone's feedback is invaluable. I do plan to add tests of course and appreciate all the suggestions.